File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
let ans = 0;
var maxDepthValue = (root,depth)=>{
    if(!root) {
        return ;
    }
    depth++;
    ans = Math.max(ans,depth);
    if(root.left)
    maxDepthValue(root.left,depth);
    if(root.right)
    maxDepthValue(root.right,depth);
}
var maxDepth = function(root) {
    
    if(!root) return 0;
    return 1 + Math.max(maxDepth(root.left),maxDepth(root.right))
    
};

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ans = 0;
    void md(TreeNode* root,int d){
        if(!root) return;
        d++;
        ans = max(ans,d);
        md(root->left,d);
        md(root->right,d);
    }
    int maxDepth(TreeNode* root) {
//         if(!root){
//             return 0 ;
//         }
        
//         //finding left subtree depth
//         int lh = maxDepth(root->left);
        
//         //finding right subtree depth
//         int rh = maxDepth(root->right);
        
//         return 1 + max(lh,rh);
        
        md(root,0);
        return ans;
    }
    
    int max(int a,int b){
        return a > b ? a : b ;
    }
};